src/lib.rs

```diff
@@ -197,23 +201,16 @@ impl Gauge {
         &self.key.labels
     }
 
-    pub fn set(&mut self, v: u64) {
-        let mut gauges = self.gauges
-            .write()
-            .expect("failed to obtain write lock for gauge");
-        if let Some(mut curr) = gauges.get_mut(&self.key) {
-            *curr = v;
-            return;
-        }
-        gauges.insert(self.key.clone(), v);
+    pub fn set(&self, v: u64) {
+        self.gauge.store(v, Ordering::Relaxed);
```

Comment:
`Relaxed` orderings always raise an eye, but given that no memory is being synchronized and it is a metric, it is probably fine. This, of course, runs the risk of the value of the gauge not matching the true last value (probably doesn't matter).